### PR TITLE
FRW-425 Refactored public documentation regarding GLUE infrastructure

### PR DESCRIPTION
**Pull request by dimitriyTsemma**
## PR Description

Ticket: https://spryker.atlassian.net/browse/FRW-493

## Checklist
- [x] I agree with the Code Contribution License Agreement in CONTRIBUTING.md

### CODE REVIEW

File: docs/scos/dev/feature-integration-guides/202204.0/glue-api/decoupled-glue-infrastructure/glue-api-documentation-generation.md [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+     */
+    protected function getContextExpanderPlugins(ContextExpanderCollectionInterface $contextExpanderCollection): ContextExpanderCollectionInterface
+    {
+        $apiApplications = [];
+        foreach ($this->getApiApplicationProviderPlugins() as $apiApplicationProviderPlugin) {
+            $apiApplications[] = $apiApplicationProviderPlugin->getName();
+        }
+        $contextExpanderCollection->addApplications($apiApplications);
+
+        $contextExpanderCollection->addExpander(new StorefrontResourcesContextExpanderPlugin(), [static::GLUE_STOREFRONT_API_APPLICATION_NAME]);
```
Comment (gechetspr) [inline docs/scos/dev/feature-integration-guides/202204.0/glue-api/decoupled-glue-infrastructure/glue-api-documentation-generation.md:220]: do we have it documented somewhere what is the second param in addExpander method and why do we have it this way in the example?

File: docs/scos/dev/glue-api-guides/202204.0/decoupled-glue-infrastructure/backend-and-storefront-api-module-differences.md [hunk 2]
```diff
@@ -117,7 +120,10 @@ use Spryker\Glue\GlueApplicationExtension\Dependency\Plugin\GlueContextExpanderP
 use Spryker\Glue\Kernel\Backend\AbstractPlugin;
 
 /**
+ * @method \Pyz\Zed\CustomApiApplication\Business\CustomApiApplicationFacadeInterface getFacade()
  * @method \Pyz\Glue\CustomApiApplication\CustomApiApplicationFactory getFactory()
+ * @method \Spryker\Zed\CustomApiApplication\Persistence\CustomApiApplicationRepositoryInterface getRepository()
```
Comment (gechetspr) [inline docs/scos/dev/glue-api-guides/202204.0/decoupled-glue-infrastructure/backend-and-storefront-api-module-differences.md:125]: Pyz

File: docs/scos/dev/glue-api-guides/202204.0/decoupled-glue-infrastructure/how-to-guides/how-to-create-or-change-a-convention.md [hunk 3]
```diff
@@ -126,3 +129,28 @@ class CustomConventionPlugin extends AbstractPlugin implements ConventionPluginI
...
+     * @return array<\Spryker\Glue\GlueApplicationExtension\Dependency\Plugin\ConventionPluginInterface>
+     */
+    protected function getConventionPlugins(): array
+    {
+        return [
+            new CustomConventionPlugin(),
+        ];
+    }
+}
+```
```
Comment (gechetspr) [inline docs/scos/dev/glue-api-guides/202204.0/decoupled-glue-infrastructure/how-to-guides/how-to-create-or-change-a-convention.md:156]: Mb we can add some paragraph with info about when we have no convention in place?

**Review by gechetspr — CHANGES_REQUESTED** [review]

**Review by gechetspr — DISMISSED** [review]

**Review by AlexSlawinski — CHANGES_REQUESTED** [review]

File: docs/scos/dev/glue-api-guides/202204.0/decoupled-glue-infrastructure/how-to-guides/how-to-create-or-change-a-convention.md [hunk 4]
```diff
@@ -126,3 +129,35 @@ class CustomConventionPlugin extends AbstractPlugin implements ConventionPluginI
     }
 }
 ```
+
+## Use a convention
+To use newly created convention you must add it to the `GlueApplicationDependencyProvider` to the method `getConventionPlugins()`.
```
[the remark this answers is not in the document]
Comment (dimitriyTsemma) [reply docs/scos/dev/glue-api-guides/202204.0/decoupled-glue-infrastructure/how-to-guides/how-to-create-or-change-a-convention.md:134]: no, it means that 'you must add it to y, which is situated inside x'
Changed to 
To use the newly created convention, you must add it to the `GlueApplicationDependencyProvider::getConventionPlugins()`.

**Review by AlexSlawinski — APPROVED** [review]

**Review by AlexSlawinski — APPROVED** [review]

**Review by AlexSlawinski — APPROVED** [review]

**Review by AlexSlawinski — APPROVED** [review]

**Pull request merged**